services/core-web/src/components/mine/Incidents/MineIncident.tsx

```diff
@@ -19,35 +19,46 @@ import { clearMineIncident } from "@common/actions/incidentActions";
 import * as Strings from "@common/constants/strings";
 import * as FORM from "@/constants/forms";
 import Loading from "@/components/common/Loading";
-import customPropTypes from "@/customPropTypes";
+// import customPropTypes from "@/customPropTypes";
 import IncidentForm from "@/components/Forms/incidents/IncidentForm";
 import ScrollSideMenu from "@/components/common/ScrollSideMenu";
 import * as routes from "@/constants/routes";
+import IMineIncident from "@mds/common";
 
-const propTypes = {
-  // eslint-disable-next-line react/no-unused-prop-types
-  incident: customPropTypes.incident.isRequired,
-  createMineIncident: PropTypes.func.isRequired,
-  fetchMineIncident: PropTypes.func.isRequired,
-  updateMineIncident: PropTypes.func.isRequired,
-  clearMineIncident: PropTypes.func.isRequired,
-  removeDocumentFromMineIncident: PropTypes.func.isRequired,
-  history: PropTypes.shape({ push: PropTypes.func, replace: PropTypes.func }).isRequired,
-  // eslint-disable-next-line react/no-unused-prop-types
-  formValues: PropTypes.objectOf(PropTypes.any).isRequired,
-  // eslint-disable-next-line react/no-unused-prop-types
-  formIsDirty: PropTypes.bool.isRequired,
-  formErrors: PropTypes.objectOf(PropTypes.string).isRequired,
-};
+export interface MineIncidentProps {
+  incident: IMineIncident;
+  createMineIncident(mineGuid: string, formattedValues: any): Promise<IMineIncident>;
```

Comment:
There is now a generic type `ActionCreator` that can be used for any of the action creators in these interfaces.  It'll cut down on having to manually add in the props/returns here.

You can add this import:

import { ActionCreator } from "@/interfaces/actionCreator";

then change these to something like this:

`createMineIncident(mineGuid: ActionCreator<typeof createMineIncident>;`

Reply:
Updated with the generic type `ActionCreator`